app/uk/gov/hmrc/customs/declaration/services/StatusResponseValidationService.scala

```diff
@@ -27,29 +28,110 @@ import scala.xml.NodeSeq
+
+  def validate(xml: NodeSeq, badgeIdentifier: BadgeIdentifier): Either[ErrorResponse, Boolean] = {
     val declarationNode = xml \ "responseDetail" \ "declarationManagementInformationResponse" \ "declaration"
-    validateBadgeIdentifier(declarationNode, badgeIdentifier) && validateReceivedDate(declarationNode)
+    val theResult =  for {
+      tradeMovementType <- extractTradeMovementType(declarationNode).right
+      xmlNode <- handleValidationForTradeTypes(badgeIdentifier, tradeMovementType, declarationNode).right
+    } yield xmlNode
+    theResult
+  }
+
+  def handleValidationForTradeTypes(badgeIdentifier: BadgeIdentifier, tradeMovementType: String, declarationNode: NodeSeq): Either[ErrorResponse, Boolean] ={
+    tradeMovementType match {
+      case EXPORT_MOVEMENT_TYPE => validateAcceptanceDate(declarationNode)
+      case _ => validateImports(badgeIdentifier, declarationNode)
+    }
+  }
+
+  private def validateImports(badgeIdentifier: BadgeIdentifier, declarationNode: NodeSeq): Either[ErrorResponse, Boolean] = {
+    val result: Either[ErrorResponse, Boolean] = for {
+      _ <- validateBadgeIdentifier(declarationNode, badgeIdentifier).right
+      _ <- validateAcceptanceDate(declarationNode).right
+    } yield true
+    result
   }
 
-  private def validateBadgeIdentifier(declarationNode: NodeSeq, badgeIdentifier: BadgeIdentifier): Boolean = {
-    extractField(declarationNode, "communicationAddress").fold(false)({ communicationAddress =>
-      badgeIdentifier.value == extractBadgeIdentifier(communicationAddress.head)
+  private def validateBadgeIdentifier(declarationNode: NodeSeq, badgeIdentifier: BadgeIdentifier): Either[ErrorResponse, Boolean] = {
+    extractField(declarationNode, "communicationAddress").fold[Either[ErrorResponse, Boolean]]({
+      declarationsLogger.errorWithoutRequestContext("Status response BadgeId field is missing")
+      Left(ErrorResponse.errorBadRequest("Badge Identifier is missing or invalid"))
+    })({ communicationAddress =>
+      if(badgeIdentifier.value != extractBadgeIdentifier(communicationAddress.head)){
+        Left(ErrorResponse.errorBadRequest("Badge Identifier is missing or invalid"))
+      }else{
+        Right(true)
+      }
     })
   }
 
-  private def extractBadgeIdentifier(communicationAddress: String) = {
-    communicationAddress.split(":").last
+  private def validateAcceptanceDate(declarationNode: NodeSeq): Either[ErrorResponse, Boolean] = {
+    extractField(declarationNode, "acceptanceDate").fold[Either[ErrorResponse, Boolean]]({
+      declarationsLogger.errorWithoutRequestContext("Status response acceptanceDate field is missing")
+      Left(ErrorResponse.errorBadRequest(s"Declaration acceptance date is greater than ${declarationsConfigService.declarationsConfig.declarationStatusRequestDaysLimit} days old"))
+    })(acceptanceDate => {
+      val parsedDateTime = ISODateTimeFormat.dateTimeNoMillis.withZoneUTC().parseDateTime(acceptanceDate.head)
```

Comment:
If the date is not the correct format it will fail with some parsing error. Should this be catered for?